src/modules/dashboard/components/TokenMintDialog/TokenMintDialog.tsx

```diff
@@ -1,113 +1,118 @@
+import * as yup from 'yup';
+import { useHistory } from 'react-router-dom';
+import { bigNumberify } from 'ethers/utils';
+import moveDecimal from 'move-decimal-point';
 
-import Button from '~core/Button';
-import Dialog from '~core/Dialog';
-import DialogSection from '~core/Dialog/DialogSection';
-import { Annotations, Input } from '~core/Fields';
-import Heading from '~core/Heading';
-import { ColonyTokens } from '~data/index';
-import { Address } from '~types/index';
+import Dialog, { DialogProps } from '~core/Dialog';
+import { ActionForm } from '~core/Fields';
+import { ActionTypes } from '~redux/index';
+import { pipe, mapPayload, withMeta } from '~utils/actions';
+import { Colony } from '~data/index';
 import { getTokenDecimalsWithFallback } from '~utils/tokens';
+import { WizardDialogType } from '~utils/hooks';
 
 import TokenMintForm from './TokenMintForm';
 
-import styles from './TokenMintDialog.css';
-
 const MSG = defineMessages({
-  title: {
-    id: 'admin.Tokens.TokenMintDialog.dialogTitle',
-    defaultMessage: 'Mint new tokens',
-  },
-  amountLabel: {
-    id: 'admin.Tokens.TokenMintDialog.amountLabel',
-    defaultMessage: 'Amount',
+  errorAmountMin: {
+    id: 'admin.Tokens.TokenMintDialog.errorAmountMin',
+    defaultMessage: 'Please enter an amount greater than 0.',
   },
-  justificationLabel: {
-    id: 'admin.Tokens.TokenMintDialog.amountLabel',
-    defaultMessage: `Explain why you're minting more tokens (optional)`,
+  errorAmountRequired: {
+    id: 'admin.Tokens.TokenMintDialog.errorAmountRequired',
+    defaultMessage: 'Please enter an amount.',
   },
 });
 
-interface FormValues {
+export interface FormValues {
   justification: string;
   mintAmount: number;
 }
 
-interface Props {
-  cancel: () => void;
-  close: () => void;
-  nativeToken: ColonyTokens[0];
-  colonyAddress: Address;
+interface CustomWizardDialogProps {
+  prevStep?: string;
+  colony: Colony;
 }
 
+type Props = DialogProps &
+  Partial<WizardDialogType<object>> &
+  CustomWizardDialogProps;
+
+const displayName = 'dashboard.TokenMintDialog';
+
+const validationSchema = yup.object().shape({
+  justification: yup.string(),
+  mintAmount: yup
+    .number()
+    .required(() => MSG.errorAmountRequired)
+    .moreThan(0, () => MSG.errorAmountMin),
+});
+
 const TokenMintDialog = ({
-  colonyAddress,
+  colony: { nativeTokenAddress, tokens = [], colonyAddress, colonyName },
+  colony,
   cancel,
   close,
-  nativeToken: { name, symbol, decimals },
-  nativeToken,
-}: Props) => (
-  <Dialog cancel={cancel}>
-    <TokenMintForm
-      colonyAddress={colonyAddress}
-      nativeToken={nativeToken}
+  callStep,
+  prevStep,
+}: Props) => {
+  const history = useHistory();
+
+  const nativeToken =
+    tokens && tokens.find(({ address }) => address === nativeTokenAddress);
+
+  const transform = useCallback(
+    pipe(
+      mapPayload(({ mintAmount: inputAmount }) => {
+        // Find the selected token's decimals
```

Comment:
This comment is wrong. You don't **find** them, you're just converting the value to `wei`